test/test-agent-stopped.js

```diff
@@ -21,9 +21,14 @@ var http = require('http');
 
 describe('test-agent-stopped', function() {
   var agent;
-  before(function() {
+  before(function(done) {
     delete process.env.GCLOUD_PROJECT;
     agent = require('..').start();
+    // Wait 300ms for agent to fail getting remote project id.
```

Comment:
I am not aware of a way to have a callback fire after the agent has resolved the metadata whether or not it resolves using `nock`.